packages/web_benchmarks/lib/analysis.dart

```diff
@@ -0,0 +1,142 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:collection/collection.dart';
+import 'server.dart';
+
+export 'src/benchmark_result.dart';
+
+/// Returns the average of the benchmark results in [results].
+///
+/// Each element in [results] is expected to have identical benchmark names and
+/// metrics; otherwise, an [Exception] will be thrown.
+BenchmarkResults computeAverage(List<BenchmarkResults> results) {
+  if (results.isEmpty) {
+    throw Exception('Cannot take average of empty list.');
+  }
+
+  BenchmarkResults totalSum = results.first;
+  for (int i = 1; i < results.length; i++) {
+    final BenchmarkResults current = results[i];
+    totalSum = totalSum._sumWith(current);
+  }
+
+  final Map<String, List<Map<String, Object?>>> average = totalSum.toJson();
+  for (final String benchmark in totalSum.scores.keys) {
+    final List<BenchmarkScore> scoresForBenchmark = totalSum.scores[benchmark]!;
+    for (int i = 0; i < scoresForBenchmark.length; i++) {
+      final BenchmarkScore score = scoresForBenchmark[i];
+      final double averageValue = score.value / results.length;
+      average[benchmark]![i][BenchmarkScore.valueKey] = averageValue;
+    }
+  }
+  return BenchmarkResults.parse(average);
+}
+
+/// Computes the delta between [test] and [baseline], and returns the results
+/// as a JSON object where each benchmark score entry contains a new field
+/// 'delta' with the metric value comparison.
+Map<String, List<Map<String, Object?>>> computeDelta(
```

Comment:
Coupling the API to JSON seems to break separation of concerns. JSON is just a serialization format. Can we wrap the result into a regular strongly-typed object and then give it a `toJson` method for cases when you just want to save it to file or send over the network?

Reply:
Added an optional field `delta` to `BenchmarkScore` to use the strongly-typed objects. PTAL.